src/fidesops/util/nested_utils.py

```diff
@@ -0,0 +1,40 @@
+from collections import defaultdict
+from typing import Dict, Any
+
+from fidesops.common_exceptions import FidesopsException
+
+
+def unflatten_dict(input_row: Dict[str, Any], separator: str = ".") -> Dict[str, Any]:
+    """
+    Takes a dictionary that has been normalized to level 1 and reconstitutes:
+
+    {
+       "a": 1,
+       "b.c": 2,
+       "b.d": 3
+    }
+
+    and turns it into:
+    {"a": 1, "b": {"c": 2, "d": 3}
+    """
+
+    def _create_dict() -> defaultdict:
+        """Can create a defaultdict at every level"""
+        return defaultdict(_create_dict)
+
+    unpacked_results = _create_dict()
+
+    for key, value in input_row.items():
+        levels = key.split(separator)
+        subdict = unpacked_results
+        for level in levels[:-1]:
+            subdict = subdict[level]
+
+        try:
+            subdict[levels[-1]] = value
```

Comment:
Thanks for the comment!

I think an IndexError is unlikely, as accessing the `-1` index always returns the last element in the list.  This could fail if the list `levels` was empty.  But levels is formed when we split the dictionary key on the separator - even something like ` "".split('.')` returns a list with one element, an empty string.

The only example I've thought of is `unflatten_dict({"": "hi"}, separator=None)`. We do specify in the signature that separator shouldn't be None, and the key of that dict is an empty string.

So maybe we leave this IndexError in, I think if this occurred, unflatten_dict would be used in a way it wasn't intended. I could be missing something, too. 

At any rate, I've added more tests to the unflatten_dict examples, and added a condition requiring that this is only run on completely flattened dicts, not partially flattened dicts to avoid surprising results.